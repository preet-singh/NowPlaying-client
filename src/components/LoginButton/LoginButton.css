@media screen and (min-width: 320px){

    .LoginButton {
        padding-top: 25px;
        width: 100%;
        text-align: right;
        padding-right: 25px;
    }

    .LoginButton a {
        color: white;
        text-decoration: none;
    }

    .LoginButton a:visited {
        color: white;
        text-decoration: none;
    }
}

@media screen and (min-width: 768px){

    #logout_button {
        position: absolute;
        top: -125px;
        right: 40px;
        color: white;
        background-color: transparent;
        border: none;
        font-size: 16px;
    }

    #logout_button:hover {
        text-decoration: underline;
        cursor: pointer;
    }

}