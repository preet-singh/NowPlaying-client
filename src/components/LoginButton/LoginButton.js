//Dependencies
import React from 'react';
import {Link} from 'react-router-dom';
import TokenService from '../../utils/token-service';

function LoginButton() {
  if(TokenService.hasAuthToken()){
    return (
      <Link to='/'>
        <button onClick={() => TokenService.clearAuthToken()}>
          Log out
        </button>
      </Link>
    )
  }
  
  else{
    return(
      <div className="LoginButton">
        <Link to="/login"><img src="#" alt="Login button" /></Link>
      </div>
    );
  }
}

export default LoginButton;