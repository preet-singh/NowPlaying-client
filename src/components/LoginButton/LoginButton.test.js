import React from 'react';
import ReactDOM from 'react-dom';
import LoginButton from './LoginButton';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json'



describe('Login button testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><LoginButton /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <LoginButton />
            </MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

})