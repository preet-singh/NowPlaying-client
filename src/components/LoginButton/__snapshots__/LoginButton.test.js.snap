// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login button testing renders UI as expected 1`] = `
<div
  className="LoginButton"
>
  <a
    href="/login"
    onClick={[Function]}
  >
    <button
      type="button"
    >
      LOGIN
    </button>
  </a>
</div>
`;

exports[`Login button testing renders empty given no tabs 1`] = `
<div
  className="LoginButton"
>
  <Link
    to="/login"
  >
    <button
      type="button"
    >
      LOGIN
    </button>
  </Link>
</div>
`;
